sympy/solvers/diophantine.py

```diff
@@ -875,9 +875,9 @@ def _diop_quadratic(var, coeff, t):
                     + (sqa*g*u**2 + D*u + sqa*F) // _c
 
                 for z0 in range(0, abs(_c)):
-                    if divisible(
-                            sqa*g*z0**2 + D*z0 + sqa*F,
-                            _c):
+                    _temp = sqa*g*z0**2 + D*z0 + sqa*F
+                    # Check if the values of y and x obtained are integers or not
+                    if divisible(_temp, _c) and divisible(_temp*e*sqc - z0*_c, sqa*_c):
```

Comment:
The constant term of x is `-(e*sqc*g*u**2 + E*u + e*sqc*F) // _c`. It is always rounded to an integer but it is correct if and only if `e*sqc*g*u**2 + E*u + e*sqc*F` is divisible by `_c`. I think the code should explicitly show this to make it more readable. It is not so clear from the condition `divisible(_temp*e*sqc - z0*_c, sqa*_c)`.